homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,243 @@
+def async_setup(hass: HomeAssistant) -> None:
+    """Set up local media source."""
+    hass.http.register_view(SynologyDsmMediaView(hass))
+
+
+class SynologyPhotosMediaSource(MediaSource):
+    """Provide Synology Photos as media sources."""
+
+    name = "Synology Photos"
+
+    def __init__(self, hass: HomeAssistant, entries: list[ConfigEntry]) -> None:
+        """Initialize Synology source."""
+        super().__init__(DOMAIN)
+        self.hass = hass
+        self.entries = entries
+
+    @property
+    def diskstation(self) -> SynologyDSMData | None:
+        """Return the Synology dsm."""
+        return self.hass.data.get(DOMAIN)
+
+    async def async_browse_media(
+        self,
+        item: MediaSourceItem,
+    ) -> BrowseMediaSource:
+        """Return media."""
+        if not self.hass.data.get(DOMAIN):
+            raise Unresolvable("Diskstation not initialized")
+        return BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=None,
+            media_class=MediaClass.DIRECTORY,
+            media_content_type=MediaClass.IMAGE,
+            title="Synology Photos",
+            can_play=False,
+            can_expand=True,
+            children_media_class=MediaClass.DIRECTORY,
+            children=[
+                *await self._async_build_diskstations(item),
+            ],
+        )
+
+    async def _async_build_diskstations(
+        self, item: MediaSourceItem
+    ) -> list[BrowseMediaSource]:
+        """Handle browsing different diskstations."""
+        ret = []
+
+        if not item.identifier:
+            for entry in self.entries:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=entry.unique_id,
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=f"{entry.title} - {entry.unique_id}",
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+            return ret
+        identifier_parts = item.identifier.split("/")
+        diskstation: SynologyDSMData = self.hass.data[DOMAIN][identifier_parts[0]]
+
+        if len(identifier_parts) == 1:
+            # Get Albums
+            # The library works sync, and this expects async calls
+            try:
+                albums = await diskstation.api.photos.get_albums()
+            except SynologyDSMException:
+                return []
+
+            ret = []
+            ret.append(
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"{item.identifier}/0",
+                    media_class=MediaClass.DIRECTORY,
+                    media_content_type=MediaClass.IMAGE,
+                    title="All images",
+                    can_play=False,
+                    can_expand=True,
+                )
+            )
```

Comment:
```suggestion
            ret = [
                BrowseMediaSource(
                    domain=DOMAIN,
                    identifier=f"{item.identifier}/0",
                    media_class=MediaClass.DIRECTORY,
                    media_content_type=MediaClass.IMAGE,
                    title="All images",
                    can_play=False,
                    can_expand=True,
                )
            ]
```

just direct start the `ret` list with its first entry